source/libraries/ContractCompiler.ts

```diff
@@ -0,0 +1,141 @@
+                errors += error.formattedMessage + "\n";
+            }
+
+            if (errors.length > 0) {
+              if (errors.match(/errors?:/i)) {
+                throw new Error("The following errors/warnings were returned by solc:\n\n" + errors);
+              } else {
+                console.warn("The following warnings were returned by solc\n\n" + errors);
+              }
+            }
+        }
+
+        // Create output directory (if it doesn't exist)
+        await asyncMkdirp(path.dirname(this.configuration.contractOutputPath));
+
+        // Output contract data to single file
+        const filteredCompilerOutput = this.filterCompilerOutput(compilerOutput);
+        await fs.writeFile(this.configuration.contractOutputPath, JSON.stringify(filteredCompilerOutput, null, '\t'));
+
+        // Output abi data to a single file
+        const abiOutput = this.generateAbiOutput(filteredCompilerOutput);
+        await fs.writeFile(this.configuration.abiOutputPath, JSON.stringify(abiOutput, null, '\t'));
+
+        return filteredCompilerOutput;
+    }
+
+    public async generateCompilerInput(): Promise<CompilerInput> {
+        const ignoreFile = function(file: string, stats: fs.Stats): boolean {
+            return file.indexOf("legacy_reputation") > -1 || (stats.isFile() && path.extname(file) !== ".sol");
+        }
+        const filePaths = await recursiveReadDir(this.configuration.contractSourceRoot, [ignoreFile]);
+        const filesPromises = filePaths.map(async filePath => (await readFile(filePath)).toString('utf8'));
+        const files = await Promise.all(filesPromises);
+
+        let inputJson: CompilerInput = {
+            language: "Solidity",
+            settings: {
+                optimizer: {
+                    enabled: true,
+                    runs: 500
+                },
+                outputSelection: {
+                    "*": {
+                        "*": [ "abi", "evm.bytecode.object" ]
+                    }
+                }
+            },
+            sources: {}
+        };
+        for (var file in files) {
+            const filePath = filePaths[file].replace(this.configuration.contractSourceRoot, "").replace(/\\/g, "/").replace("/src", "");
+            if (filePath.includes(".t.sol") || filePath.includes("test.sol")) continue;
+            inputJson.sources[filePath] = { content : files[file] };
+        }
+
+        return inputJson;
+    }
+
+    private filterCompilerOutput(compilerOutput: CompilerOutput): CompilerOutput {
+        const result: CompilerOutput = { contracts: {} };
+        for (let relativeFilePath in compilerOutput.contracts) {
+          let contractsInFile = compilerOutput.contracts[relativeFilePath];
+          for (let contractName in contractsInFile) {
+                if (!this.configuration.activeContracts.includes(relativeFilePath)) continue;
```

Comment:
Use whitelist

Reply:
Ah, is `activeContracts` something you added?  If so, can you give an overview of what it is for?

Reply:
We are compiling a lot of contracts we don't care about, directly. Every ds-* contract, which is a dependency of sai, but won't be called directly. This just allows the configuration to state which files we care about. There was similar logic in here for augur-core, but it was more "deploy everything, use various path and naming conventions to see if it is a library"